cmd/kube-proxy/app/server.go

```diff
@@ -653,7 +653,9 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 // CleanupAndExit remove iptables rules and exit if success return nil
 func (s *ProxyServer) CleanupAndExit() error {
 	encounteredError := userspace.CleanupLeftovers(s.IptInterface)
-	encounteredError = iptables.CleanupLeftovers(s.IptInterface) || encounteredError
+	encounteredError = iptables.FlushIptablesChains(s.IptInterface) || encounteredError
+	encounteredError = ipvs.FlushIptablesChains(s.IptInterface) || encounteredError
+	encounteredError = iptables.DeleteIptablesChains(s.IptInterface) || encounteredError
 	encounteredError = ipvs.CleanupLeftovers(s.IpvsInterface, s.IptInterface, s.IpsetInterface, s.CleanupIPVS) || encounteredError
```

Comment:
Rename ipvs.CleanupLeftovers also to be consistent?

Reply:
ipvs.CleanupLeftovers clean ipvs rules, dummy interface, iptables chains and ipset, so we should separate ipvs.FlushIptablesChains form ipvs.CleanupLeftovers ?

Reply:
or ipvs.CleanupLeftovers -> ipvs.FluashIPVS + ipvs.DeleteDummyDevice + ipvs. DeleteIptablesChains + ipvs.DeleteIPSet ?

Reply:
ah I see.. let's leave that as is then but can we rename `iptables.DeleteIptablesChains` back to `iptables.CleanupLeftovers`? Just keeps things consistent and easier to follow: "for every proxy we do "Flush" then "CleanupLeftovers"

Reply:
ok